### PR TITLE
Require Python 3.6 and update Azure configuration

### PR DESCRIPTION
**Pull request by astrofrog**
This will now test against recent Sphinx versions

### CODE REVIEW

**Comment by astrofrog** [comment]
@eteq @pllim @bsipocz - I plan to release a new version tomorrow, so thought it would make sense to update the testing config and drop Python 2.

File: tox.ini [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
+    sphinx24: sphinx==2.4.*
+    sphinx30: sphinx==3.0.*
+    sphinx31: sphinx==3.1.*
+    sphinx32: sphinx==3.2.*
     sphinxdev: git+https://github.com/sphinx-doc/sphinx.git
 extras =
     test: test
 commands =
-    test: pytest --pyargs sphinx_automodapi --cov-append --cov sphinx_automodapi --cov-report=
+    test: pytest --pyargs sphinx_automodapi --cov sphinx_automodapi
```
Comment (pllim) [inline tox.ini:23]: Running coverage for everything seems overkill.
```suggestion
    test: pytest --pyargs sphinx_automodapi
```

**Review by pllim — COMMENTED** [review]
LGTM, except the bit about coverage.

File: tox.ini (hunk 1 above)
Comment (astrofrog) [reply tox.ini:23]: It's pretty cheap for a small package like this though - any reason why not to do it?

Comment (pllim) [reply tox.ini:23]: We don't seem to be reporting coverage. So, what is the point?

Comment (astrofrog) [reply tox.ini:23]: Ah of course then I agree! But really we should report coverage, will try and fix in a follow up PR

**Review by pllim — APPROVED** [review]
:shipit:  Thanks!

**Pull request merged**